scriptmodules/archivefuncs.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+## @file archivefuncs.sh
+## @brief RetroPie archivefuncs library
+## @copyright GPLv3
+
+readonly arch_dir="/tmp/retropie_archive"
+
+function __zipExtract() {
+    unzip "$1" -d "$2"
+}
```

Comment:
It's likely to be a one liner per archive format or not much more, so no need for functions - just do a case statement for checking the extension and put it in there.